integrations/audio.cpp

```diff
@@ -21,101 +19,219 @@ class Audio : public QObject
-    //TODO implement sensors/selector for sink/source
+    Number *m_sourceVolume = nullptr;
+    Select *m_sinkSelector = nullptr;
+    Select *m_sourceSelector = nullptr;
+    
     PulseAudioQt::Sink *m_sink = nullptr;
-
+    PulseAudioQt::Source *m_source = nullptr;
+    PulseAudioQt::Context *m_ctx = nullptr;
 };
 
 // Constructor
 Audio::Audio(QObject *parent)
     : QObject(parent)
 {
     m_sinkVolume = new Number(this);
-    m_sinkVolume->setId("volume");
-    m_sinkVolume->setName("System Volume");
+    m_sinkVolume->setId("output_volume");
+    m_sinkVolume->setName("Output Volume");
+    m_sinkVolume->setDiscoveryConfig("icon", "mdi:knob");
     m_sinkVolume->setRange(0, 100, 1, "%");
 
     connect(m_sinkVolume, &Number::valueChangeRequested,
-            this, &Audio::setVolume);
+            this, &Audio::setSinkVolume);
+
+    m_sourceVolume = new Number(this);
+    m_sourceVolume->setId("input_volume");
+    m_sourceVolume->setName("Input Volume");
+    m_sourceVolume->setDiscoveryConfig("icon","mdi:microphone");
+    m_sourceVolume->setRange(0, 100, 1, "%");
+
+    connect(m_sourceVolume, &Number::valueChangeRequested,
+            this, &Audio::setSourceVolume);
+
 
-    auto ctx = PulseAudioQt::Context::instance();
-    if (!ctx || !ctx->isValid()) {
+    m_ctx = PulseAudioQt::Context::instance();
+    if (!m_ctx || !m_ctx->isValid()) {
         qWarning() << "Audio: PulseAudio context not valid";
         return;
     }
 
-    auto *m_server = ctx->server();
-    if (!m_server) {
+    auto *server = m_ctx->server();
+    if (!server) {
         qWarning() << "Audio: No PulseAudio server";
         return;
     }
 
-    connect(m_server, &PulseAudioQt::Server::defaultSinkChanged,
-            this, &Audio::onDefaultSinkChanged);
-
-    auto *initial = m_server->defaultSink();
-    if (initial) {
-        onDefaultSinkChanged(initial);
-    } else {
-        qWarning() << "Audio: default sink not ready, waiting for signal";
-    }
+    // Sink selctor and signal connection
+    m_sinkSelector = new Select(this);
+    m_sinkSelector->setId("volume_output_selector");
+    m_sinkSelector->setDiscoveryConfig("icon","mdi:volume-source");
+    m_sinkSelector->setName("Output Device");
+    connect(m_sinkSelector, &Select::optionSelected,
+            this, &Audio::onSinkSelected);
+    //Microphone selector and signal connection
+    m_sourceSelector = new Select(this);
+    m_sourceSelector->setId("volume_input_selector");
+    m_sourceSelector->setDiscoveryConfig("icon","mdi:microphone-settings");
+    m_sourceSelector->setName("Input Device");
+    connect(m_sourceSelector, &Select::optionSelected,
+            this, &Audio::onSourceSelected);
+
+    // Connect to signal when default sink changes
+    connect(server, &PulseAudioQt::Server::defaultSinkChanged,
+            this, &Audio::updateSinks);
+
+    // Connect to signal when default source changes
+    connect(server, &PulseAudioQt::Server::defaultSourceChanged,
```

Comment:
presumably we also need to call updateSinks/updateSources when a non-default sink/source is added/removed.